static/app/chartcuterie/discover.tsx

```diff
@@ -173,7 +173,10 @@ discoverCharts.push({
 
     const stats = Object.values(data.stats);
     const hasOther = Object.keys(data.stats).includes('Other');
-    const color = theme.charts.getColorPalette(stats.length - 2 - (hasOther ? 1 : 0));
+    const color = theme.charts
+      .getColorPalette(stats.length - 2 - (hasOther ? 1 : 0))
+      ?.slice() as string[];
```

Comment:
slice() at first looked weird to me. I would've done `[...foo]`

If there any way to write it so we don't have to manually add the `string[]` type back in?

Also: does the `?` in there meant that the type should be `string[] | undefined`?

Reply:
Ah, yes this isn't precise. Let me see if I can remove the cast entirely

Reply:
There is no validation later on, so I'm going to add a nullish coalescing before casting